docs/src/cli/delegate-stake.md

```diff
@@ -0,0 +1,134 @@
+```
+
+`<STRING>` is arbitrary but typically a number corresponding to which derived
+account this is. The first account might be "0", then "1", and so on.
+`<KEYPAIR>` is the same for both the source keypair and account keypair,
+because the second instance is used only as a base address. The command
+derives a new address from the base address and seed string. To see what stake
+address the command will derive, use `solana create-address-with-seed`:
+
+```bash
+solana create-address-with-seed --from=<PUBKEY> <SEED_STRING> STAKE
+```
+
+It will output a derived public key, which can be used for the
+`<ACCOUNT_PUBKEY>` argument in staking operations.
+
+## Set Stake and Withdraw Authorities
+
+Staking commands look to keypairs to authorize certain stake account
+operations. It uses a stake authority to authorize stake delegation,
+deactivating stake, splitting stake, and setting a new stake authority.  It
+uses a withdraw authority to authorize withdrawing stake, and setting either
+a new stake or withdraw authority.
+
+Stake and withdraw authorities can be set when creating account via the
+`--stake-authority` and `--withdraw-authority` options, or afterward with
+the `solana stake-authorize` command. For example, to set a new authorized
+staker, run:
+
+```bash
+solana stake-authorize <ACCOUNT_KEYPAIR> --stake-authority=<KEYPAIR> --new-stake-authority=<PUBKEY>
+```
+
+This will use the existing stake authority `<KEYPAIR>` to authorize a new
+stake authority `<PUBKEY>` on the stake account `<ACCOUNT_KEYPAIR>`.
+
+## Delegate Stake
+
+To delegate stake, run:
+
+```bash
+solana delegate-stake --stake-authority=<KEYPAIR> <ACCOUNT_PUBKEY> <VOTE_PUBKEY>
+```
+
+`<KEYPAIR>` authorizes the operation on the account with address
+`<ACCOUNT_PUBKEY>`. The stake is delegated to the vote account with
+public key `<VOTE_PUBKEY>`.
+
+## Deactivate Stake
+
+Once delegated, you can deactivate stake with the `solana deactivate-stake`
+command:
+
+```bash
+solana deactivate-stake --stake-authority=<KEYPAIR> <ACCOUNT_PUBKEY>
+```
+
+`<KEYPAIR>` authorizes the operation on the account with address
+`<ACCOUNT_PUBKEY>`.
+
+Note that stake takes several epochs to "cool down". Attempts to delegate
+stake in the cooldown period will fail.
+
+## Split Stake
+
+All stake in a stake account is delegated to a single validator. If you'd
+prefer to spread your stake over multiple validators use the
+`solana split-stake` command to duplicate stake accounts and split your
+tokens between them:
+
+```bash
+solana split-stake --stake-authority=<KEYPAIR> <ACCOUNT_PUBKEY> <NEW_ACCOUNT_KEYPAIR> <AMOUNT>
+```
+
+`<ACCOUNT_PUBKEY>` is the existing stake account, `<KEYPAIR>` is the
+authorized staker, `<NEW_ACCOUNT_KEYPAIR>` is the keypair for the new account,
+and `<AMOUNT>` is the number of tokens to transfer to the new account.
+
+## Withdraw Stake
+
+Transfer tokens out of a stake account with the `solana withdraw-stake` command:
```

Comment:
This section should mention that withdraws are subject to both cool down and lockup

Reply:
Do you know the details of that, by chance? I'd have to go sift through the code.

Reply:
Withdraw eligibility seems to be
- `0` if lockup is in force
- `0` if in warm up or fully staked
- `cooled_amount` if in cool down
- `balance` or `balance - rent_exempt` if undelegated, but initialized account
- `balance` if the account is uninitialized
Should be a piece of cake to distill that down! :upside_down_face:

Reply:
Okay, cool. I'm going to do warmup/cooldown details in a followup PR. This might require another page.

Reply:
We should probably look into refactoring the code such that we can reuse part of it to calculate an "eligible for withdraw" value for display in `stake-account`